### PR TITLE
Close #1719 allow customer to have recurring promotion

### LINKED ISSUE #1719: allow customer to have recurring promotion

### PR DESCRIPTION
**Pull request by tykealy**
### Promotion will be applied every month.
image
image

### CODE REVIEW

File: spec/interactors/spree_cm_commissioner/customer_promotion_creator_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
+require 'spec_helper'
+
+RSpec.describe SpreeCmCommissioner::CustomerPromotionCreator do
+  let(:customer) { create(:cm_customer) }
+  let(:store) { create(:store) }
+  describe '.call' do
+    context "when customer has no existing promotion" do
+      it "creates a new promotion" do
```
Comment (channainfo) [inline spec/interactors/spree_cm_commissioner/customer_promotion_creator_spec.rb:8]: add a case where customer is not found

**Review by channainfo — CHANGES_REQUESTED** [review]

**Comment by channainfo** [comment]
@tykealy let's rebase this with develop. the number of commits here is overwhelming

**Comment by tykealy** [comment]
> @tykealy let's rebase this with develop. the number of commits here is overwhelming

@channainfo, I've rebased with develop bong.

File: db/migrate/20240730073727_create_super_admin_role.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class CreateSuperAdminRole < ActiveRecord::Migration[7.0]
+  def change
+    super_admin = Spree::Role.find_or_create_by(name: 'super_admin')
```
Comment (channainfo) [inline db/migrate/20240730073727_create_super_admin_role.rb:3]: Remove this migration.  Do it via rails console instead.

**Review by channainfo — CHANGES_REQUESTED** [review]

**Pull request merged**